### PR TITLE
[17.0][MIG] fieldservice_sale_recurring: Migration to 17.0

### PR DESCRIPTION
**Pull request by imlopes**

### CODE REVIEW

File: fieldservice_sale_recurring/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                """This recurring has been created from: <a href=
+                   # data-oe-model=sale.order data-oe-id=%(order_id)s>%(order)s</a>
+                   (%(product)s)
+                """
+            ).format(
+                {
+                    "order_id": so_line.order_id.id,
+                    "order": so_line.order_id.name,
+                    "product": so_line.product_id.name,
+                }
```
Comment (brian10048) [inline fieldservice_sale_recurring/models/sale_order_line.py]: String format needs fixed:
Screenshot_20240916_160908


```suggestion
                """This recurring has been created from: <a href=
                   # data-oe-model=sale.order data-oe-id={order_id}>{order}</a>
                   ({product})
                """
            ).format(
                order_id = so_line.order_id.id,
                order = so_line.order_id.name,
                product = so_line.product_id.name,
            )
```

File: fieldservice_sale_recurring/models/sale_order_line.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            # post message on SO
+            msg_body = _(
+                """Field Service recurring Created (%(product)s): <a href=
+                   # data-oe-model=fsm.recurring data-oe-id=%(id)s>%(name)s</a>
+                """
+            ).format(
+                {
+                    "product": so_line.product_id.name,
+                    "id": fsm_recurring.id,
+                    "name": fsm_recurring.name,
```
Comment (brian10048) [inline fieldservice_sale_recurring/models/sale_order_line.py]:
```suggestion
                """Field Service recurring Created ({product}): <a href=
                   # data-oe-model=fsm.recurring data-oe-id={id}>{name}</a>
                """
            ).format(
                product = so_line.product_id.name,
                id = fsm_recurring.id,
                name = fsm_recurring.name,
```

**Review by brian10048 — CHANGES_REQUESTED** [review]
quick functional review and works as intended.

Need to fix string formatting

File: fieldservice_sale_recurring/models/sale_order_line.py [hunk 3]
```diff
@@ -38,7 +38,7 @@ def _field_create_fsm_recurring_prepare_values(self):
 
     def _field_create_fsm_recurring(self):
         """Generate fsm_recurring for the given so line, and link it.
-        :return a mapping with the so line id and its linked fsm_recurring
+        return a mapping with the so line id and its linked fsm_recurring
```
Comment (yankinmax) [inline fieldservice_sale_recurring/models/sale_order_line.py]: this is normal when we describe input/output parameters/variables, leave its as it's pls

**Review by yankinmax — APPROVED** [review]

File: fieldservice_sale_recurring/models/sale_order_line.py (hunk 2 above)
Comment (imlopes) [reply fieldservice_sale_recurring/models/sale_order_line.py]: image


image

File: fieldservice_sale_recurring/readme/INSTALL.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+To install Field Service and have the mapping features, you need to
+install GeoEngine.
+
+Please refer to the installation instructions available at:
+<https://github.com/OCA/geospatial/tree/13.0/base_geoengine>
```
Comment (gurneyalex) [inline fieldservice_sale_recurring/readme/INSTALL.md]: Please remove the file, it has nothing to do with the fieldservice_sale_recurring addon.

File: fieldservice_sale_recurring/readme/USAGE.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+- Go to Sales
+- Create a new Quotation/Sale Order
+- Set the FSM Location to be used
+- On a Sale Order Line, select a product configured for field service
+  orders
+- Confirm the Sale Order
+- Field Service Orders linked to SO lines are created
```
Comment (gurneyalex) [inline fieldservice_sale_recurring/readme/USAGE.md]: I don't think this explanation is the one for fieldservice_sale_recurring. Can you update the USAGE ?

**Comment by gurneyalex** [comment]
hello @brian10048 Can you update your review?

**Review by gurneyalex — COMMENTED** [review]
@imlopes Can you also squash the translation commits (see https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate)

**Comment by max3903** [comment]
@brian10048 Can you help us here?

**Review by max3903 — APPROVED** [review]

**Review by brian10048 — APPROVED** [review]

**Comment by brian10048** [comment]
Working as intended.  Thank you!

File: fieldservice_sale_recurring/models/sale_order_line.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        """
+        dom = super()._get_invoiceable_fsm_order_domain()
+        if self.fsm_recurring_id:
+            dom.append(("fsm_recurring_id", "=", self.fsm_recurring_id.id))
+        return dom
+
+    def _prepare_invoice_line(self, **optional_values):
+        res = super()._prepare_invoice_line(**optional_values)
+        if self.fsm_recurring_id:
+            fsm_orders = self._get_invoiceable_fsm_order()
```
Comment (imlopes) [inline fieldservice_sale_recurring/models/sale_order_line.py:87]: It seems this method was added here : https://github.com/OCA/field-service/commit/b6fa0fbce6e575dd46866fc9775feb2facf8dd04#diff-39a34800cd8567f4f4bc2657da63a2c24753af76b678249c2837f9ba6cc9f944 and the code was migrated (to 15.0 and here to 17.0) with this call but I couldn't find the method.

Am I missing something ?
@max3903 @gurneyalex

Comment (max3903) [reply fieldservice_sale_recurring/models/sale_order_line.py:87]: @imlopes Good question. 

@brian10048 Can you help?

**Pull request merged**